scripts/agent_step_receipt_relabeling_harness.py

```
@@ -0,0 +1,696 @@
+            raise AgentReceiptError("unknown dependency trust class")
+        if trust_class != "dependency_dropped":
+            raise AgentReceiptError("dependency-drop entry must use dependency_dropped trust class")
+        if verifier_domain != VERIFIER_DOMAIN:
+            raise AgentReceiptError("dependency-drop verifier domain mismatch")
+        if not isinstance(entry["reason_for_drop"], str) or not entry["reason_for_drop"]:
+            raise AgentReceiptError("dependency-drop reason must be a non-empty string")
+        required = entry["required_subproof_or_attestation"]
+        if required is not None:
+            required = require_dict(required, "required subproof/attestation")
+            if set(required) != {"kind", "commitment", "verifier_domain"}:
+                raise AgentReceiptError("required subproof/attestation has unexpected keys")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)